pvm.py

```diff
@@ -35,116 +43,230 @@ def _init_logger():
+				media2.pause()
 				_logger.info("%s command success.", command)
+			elif command=="pause":
+				if canPause:
+						media1.pause()
+						media2.pause()
+						pos1 = media1.position()
+						pos2 = media2.position()
+						_logger.info("media 1 position: %s", pos1)
+						_logger.info("media 2 position: %s", pos2)
+						if pos1 > pos2:
+							media1.set_position(pos2)
+							_logger.info("media 1 set position: %s", pos2)
+						elif pos1 < pos2:
+							media2.set_position(pos1)
+							_logger.info("media 2 set position: %s", pos1)
+						canPause = False
+						canStart = True
+						_logger.info("%s command success.", command)
 			else:
-				_logger.info("%s command failed.", command)
+				_logger.info("%s unknown.", command)
 		else:
-			_logger.info("%s unknown.", command)
+			# If Raspberry Pi only has one display.
+			if command=="start":
+				if canStart:							
+						media1.play()
+						canPause = True
+						canStart = False
+						_logger.info("%s command success.", command)
+				else:
+					_logger.info("%s command failed.", command)
+			elif command=="stop":
+				if media1.can_quit():
+					media1.quit()
+					IS_FILE_SET = False
+					canStart = False
+					canPause = False
+					_logger.info("%s command success and file has been unset.", command)
+				else:
+					_logger.info("%s command failed.", command)
+			elif command=="set_position":
+				media1.set_position(float(value))
+				_logger.info("%s command success.", command)
+			elif command=="set_rate":
+				fps = get_rate(VIDEO_PATH_ONE, value)
+				media1 = OMXPlayer(VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop', '--force-fps', fps])
+				media1.pause()
+				canStart = True
+				canPause = False
+				_logger.info("%s command success.", command)
+			elif command=="pause":
+				if canPause:
+						media1.pause()
+						canPause = False
+						canStart = True
+						_logger.info("%s command success.", command)
+				else:
+					_logger.info("%s command failed.", command)
+			else:
+				_logger.info("%s unknown.", command)
 	except Exception as e:
 		# `logger#exception method prints the stack trace`
 		_logger.exception("Function: parse_commands failed! %s" % (e))
 
+# get fps from video path
+def get_rate(path, value):
+	media = OMXPlayer(VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer', args=['--loop','--force-fps', fps])
+	video_info = subprocess.Popen(["omxplayer", "-i", VIDEO_PATH_ONE], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+	out, err = video_info.communicate()
+	out = out.decode(encoding='utf-8')
+	splist = out.split(", ")
+	for s in splist:
+		if 'fps' in s:
+			fps_info = s.split(' ')
+			original_fps = float(fps_info[0])
+	fps = str(original_fps * float(value))
+
+def parse_value(value):
+	value1, value2 = value.split("|")
+	if value1 is None or value2 == '':
+		value2 = value1
+	return value1, value2
```

Comment:
Here is a different way to look at it

```suggestion
def parse_value(value):
	values = [None,None]
	if "|" in value:
		values[0], values[1] = value.split("|")
		if values[1] == '':
			values[1] = values[0]
	else:
		values[0] = value
	return values
```